scripts/tools/metricTracer.yml

```diff
@@ -0,0 +1,255 @@
+type: header
+desc: "Intel $OneApi Level-Zero Tool Experimental Extension for Metrics Tracer"
+version: "1.10"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Metric Tracer Experimental Extension Name"
+version: "1.10"
+name: $T_METRICS_TRACER_EXP_NAME
+value: '"$XT_experimental_metric_tracer"'
+--- #--------------------------------------------------------------------------
+type: handle
+desc: "Handle of metric tracer's object"
+version: "1.10"
+class: $tMetricTracer
+name: "$t_metric_tracer_exp_handle_t"
+--- #--------------------------------------------------------------------------
+type: handle
+desc: "Handle of metric decoder's object"
+version: "1.10"
+class: $tMetricDecoder
+name: "$t_metric_decoder_exp_handle_t"
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Metric tracer descriptor"
+version: "1.10"
+class: $tMetricTracer
+name: $t_metric_tracer_exp_desc_t
+base: $t_base_desc_t
+members:
+    - type: uint32_t
+      name: "notifyEveryNBytes"
+      desc: >
+            [in,out] number of collected bytes after which notification event will be signaled.
+            If the requested value is not supported exactly, then the driver may use a value that
+            is the closest supported approximation and shall update this member during
+            $tMetricTracerCreateExp.
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Decoded metric entry"
+version: "1.10"
+name: $t_metric_entry_exp_t
+members:
+    - type: $t_value_t 
+      name: "value"
+      desc: "[out] value of the decodable metric entry or event. Number is meaningful based on the metric type."
+    - type: uint64_t  
+      name: "timeStamp"
+      desc: "[out] timestamp at which the event happened."
+    - type: uint32_t
+      name: "metricIndex"
+      desc: "[out] index to the decodable metric handle in the input array (phMetric) in $tMetricTracerDecode()"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Create a metric tracer for a device."
+version: "1.10"
+class: $tMetricTracer
+name: CreateExp
+decl: static
+details:
+    - "The notification event must have been created from an event pool that was created using $X_EVENT_POOL_FLAG_HOST_VISIBLE flag."
+    - "The duration of the signal event created from an event pool that was created using $X_EVENT_POOL_FLAG_KERNEL_TIMESTAMP flag is undefined. However, for consistency and orthogonality the event will report correctly as signaled when used by other event API functionality."
+    - "The application must **not** call this function from simultaneous threads with the same device handle."
+    - "The metric tracer is created in disabled state"
+    - "Metric groups must be of the sampling type ZET_METRIC_SAMPLING_TYPE_EXP_FLAG_TRACER_BASED"
```

Comment:
Metric groups can support multiple sampling types. So may be this could be reworded to mention "support ZET_METRIC_SAMPLING_TYPE_EXP_FLAG_TRACER_BASED"

Reply:
Good point! fixed